File: ws/pazmino/u3/ws40CreateRead/PIBApp/src/main/java/utils/MongoDBConnection.java
package utils;

import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import ec.edu.espe.pibapp.model.CountryGDP;
import org.bson.Document;

/**
 *
 * @author Brandon Pazmino
 */
public class MongoDBConnection {
 public static MongoClient createConnection() {
        String connectionString = "mongodb+srv://bpazmino:bpazmino@cluster0.ot5jx.mongodb.net/";
        MongoClient mongoClient = MongoClients.create(connectionString);
        return mongoClient;
    }
    
    public MongoCollection<Document> connect() {
        MongoDatabase database = createConnection().getDatabase("OOP");
        return database.getCollection("PIB");
    }

    public static boolean add(CountryGDP countryGDP) {
        MongoDBConnection mongoDBConnection = new MongoDBConnection();
        MongoCollection<Document> collection = mongoDBConnection.connect();
        boolean saved = false;

        try {
            Document doc = new Document("countryName", countryGDP.getCountryName())
                    .append("gdpPerCapita", countryGDP.getGdpPerCapita());

            collection.insertOne(doc);
            saved = true;
        } catch (Exception ex) {
            System.out.println("Error al guardar el PIB per cápita: " + ex.getMessage());
            saved = false;
        }

        return saved;
    }

    public static double calculateDifference(CountryGDP countryGDP1, CountryGDP countryGDP2) {
        return Math.abs(countryGDP1.getGdpPerCapita() - countryGDP2.getGdpPerCapita());
    }
}

File: hw/pazmino/u2/HW25GUIAndMongo/ContactsBookPython/utils/mongo_db_connection.py
from pymongo import MongoClient
from datetime import datetime
from typing import List
try:
    from ec.edu.espe.contact_book.model.contact import Contact 
except ImportError:
    from ec.edu.espe.contact_book.model.contact import Contact  

class MongoConnection:
    
    _client = None  # Cliente MongoDB como variable estática
    
    @staticmethod
    def get_database():
        """Devuelve la instancia de la base de datos OOP."""
        if MongoConnection._client is None:
            MongoConnection._client = MongoClient("mongodb+srv://bpazmino:bpazmino@cluster0.ot5jx.mongodb.net/")
        return MongoConnection._client["OOP"]

class MongoDbManager:
    
    @staticmethod
    def connect():
        database = MongoConnection.get_database()
        return database["CONTACTSBOOK"]

    @staticmethod
    def add(contact):
    
        collection = MongoDbManager.connect()
        
        try:
            if isinstance(contact, Contact):
                collection.insert_one(contact.to_dict())  # Usa el método to_dict() de Contact
                return True
        except Exception as ex:
            print(f"Error al guardar el contacto: {ex}")
        
        return False
